Fix timing out RemoteObjectRegistry API test

ronan-apple opened the pull request:
#### b960368ecc633d16d387bcc97d66d2f09aaeda2c
<pre>
Fix timing out RemoteObjectRegistry API test
<a href="https://bugs.webkit.org/show_bug.cgi?id=297294">https://bugs.webkit.org/show_bug.cgi?id=297294</a>
<a href="https://rdar.apple.com/158168034">rdar://158168034</a>

Reviewed by Alex Christensen.

Update coreipc.js for API tests which fixes the CallReplyBlockWithBadInvocation
test which was timing out due to failing to parse the RemoteObjectRegistry message.

This brings coreipc.js in TestWebKitAPI back in line with LayoutTests, and fixes
a bug in both in parseKeyValuePair.

* LayoutTests/ipc/coreipc.js:
(CoreIPCClass.prototype.initializeMessages):
(CoreIPCClass.prototype.generateSendingFunction):
(export.StreamConnectionInterface.prototype.initializeMessages):
(export.StreamConnectionInterface.prototype.generateStreamSendingFunction):
(export.StreamConnectionInterface):
(export.resolveAlias):
(export.ArgumentSerializer.splitTemplateType):
(export.ArgumentSerializer.parseTemplate):
(export.ArgumentSerializer):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/coreipc.js:
(CoreIPCClass.prototype.initializeMessages):
(CoreIPCClass.prototype.generateSendingFunction):
(StreamConnectionInterface.prototype.initializeMessages):
(StreamConnectionInterface.prototype.generateStreamSendingFunction):
(StreamConnectionInterface):
(resolveAlias):
(ArgumentSerializer.splitTemplateType):
(ArgumentSerializer.parseTemplate):
(ArgumentSerializer):

Canonical link: <a href="https://commits.webkit.org/298685@main">https://commits.webkit.org/298685@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2d54b26063489f18fc80cb09f4c718ce0e3a9835

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116328 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35989 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26538 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122385 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66889 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/28afc8ab-c83c-49cf-8c94-4eae805e2db8) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118217 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36683 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44577 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88356 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42847 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/564a4958-57c4-44f4-9549-3e2012b10965) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119277 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29260 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104372 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68788 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28342 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22474 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66066 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98642 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22636 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125534 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43222 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32459 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97063 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43587 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100578 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96859 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42151 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20047 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/39172 "The change is no longer eligible for processing.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18578 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43109 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42576 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45916 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44281 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/coreipc.js:
```diff
@@ -212,19 +212,13 @@ const aliases = {
     'unsigned long': 'uint64_t',
     'long': 'uint64_t',
     'unsigned char': 'uint8_t',
     'CGFloat': 'double',
```
You should be able to remove this too.  Same with NSUInteger, NSInteger, long, unsigned long, and size_t.

ronan-apple replied:
Ah, ace, I didn't realise this - did you just know these types are not used? Re-ran the tests that rely on CoreIPC.js just to check, seems good!

achristensen07 commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/coreipc.js:
```diff
@@ -380,6 +380,13 @@ class ArgumentSerializer {
                 count += 1;
                 result.push([ ArgumentSerializer.serializeArgument(argumentDefinition, element) ]);
             }
+            if(isStdSpan && splitType.length > 1) {
```
WebKit style requires a space between if and (, and I think that should apply here too.

ronan-apple replied:
Sure thing! Corrected this across both files.

achristensen07 commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/coreipc.js:
```diff
@@ -900,14 +971,61 @@ class ArgumentParser {
         return [position, values];
     }
+    /*
+    static serializeStdArray(innerType, argument) {
```
Let's not check in commented-out code.

ronan-apple replied:
Good point, removed!

Commit:
Fix timing out RemoteObjectRegistry API test
https://bugs.webkit.org/show_bug.cgi?id=297294
rdar://158168034

Reviewed by Alex Christensen.

Update coreipc.js for API tests which fixes the CallReplyBlockWithBadInvocation
test which was timing out due to failing to parse the RemoteObjectRegistry message.

This brings coreipc.js in TestWebKitAPI back in line with LayoutTests, and fixes
a bug in both in parseKeyValuePair.

* LayoutTests/ipc/coreipc.js:
(CoreIPCClass.prototype.initializeMessages):
(CoreIPCClass.prototype.generateSendingFunction):
(export.StreamConnectionInterface.prototype.initializeMessages):
(export.StreamConnectionInterface.prototype.generateStreamSendingFunction):
(export.StreamConnectionInterface):
(export.resolveAlias):
(export.ArgumentSerializer.splitTemplateType):
(export.ArgumentSerializer.parseTemplate):
(export.ArgumentSerializer):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/coreipc.js:
(CoreIPCClass.prototype.initializeMessages):
(CoreIPCClass.prototype.generateSendingFunction):
(StreamConnectionInterface.prototype.initializeMessages):
(StreamConnectionInterface.prototype.generateStreamSendingFunction):
(StreamConnectionInterface):
(resolveAlias):
(ArgumentSerializer.splitTemplateType):
(ArgumentSerializer.parseTemplate):
(ArgumentSerializer):

Canonical link: https://commits.webkit.org/298685@main
